configure.ac

```diff
@@ -27,4 +27,34 @@ AC_SUBST(LTLIBOBJS)
 
 AC_CONFIG_FILES(Makefile libudffs/Makefile mkudffs/Makefile cdrwtool/Makefile pktsetup/Makefile udffsck/Makefile wrudf/Makefile doc/Makefile)
 
+AC_ARG_ENABLE(debug,
+AS_HELP_STRING([--enable-debug],
+[enable debugging, default: no]),
+[case "${enableval}" in
+    yes) debug=true ;;
+    no)  debug=false ;;
+    *)   AC_MSG_ERROR([bad value ${enableval} for --enable-debug]) ;;
+esac],
+[debug=false])
+
+AM_CONDITIONAL(DEBUG, test x"$debug" = x"true")
+
+AC_ARG_ENABLE(tests,
+AS_HELP_STRING([--enable-tests],
+[enable tests building, default: no]),
+[case "${enableval}" in
+    yes) tests=true ;;
+    no)  tests=false ;;
+    *)   AC_MSG_ERROR([bad value ${enableval} for --enable-tests]) ;;
+esac],
+[tests=false])
+
+AM_CONDITIONAL(TESTS, test x"$tests" = x"true")
+
+AC_C_BIGENDIAN
+AM_CONDITIONAL(WORDS_LITTLEENDIAN, test "x$ac_cv_c_bigendian" = "xno")
+AM_CONDITIONAL(WORDS_BIGENDIAN, test "x$ac_cv_c_bigendian" = "xyes")
+
+AC_PROG_CC_C99
```

Comment:
If C99 is really needed, then put it after AC_PROG_CC macro and properly check if compiler is C99. This macro itself does not throw any error, just set variable ac_cv_prog_cc_c99.

Reply:
Clean-up was made here:  f2ce3c2

Reply:
I believe this should do the check: AC_PROG_CC_STDC.

> Macro: AC_PROG_CC_STDC
> 
> If the C compiler cannot compile ISO Standard C (currently C99), try to add an option to output variable CC to make it work. If the compiler does not support C99, fall back to supporting ANSI C89 (ISO C90).
>
> After calling this macro you can check whether the C compiler has been set to accept Standard C; if not, the shell variable ac_cv_prog_cc_stdc is set to ‘no’. 

So after then checking ac_cv_prog_cc_stdc and throwing  error should be enough, right?

Reply:
If your code depends on C99 syntax, then you need to use AC_PROG_CC_C99.
https://www.gnu.org/software/autoconf/manual/autoconf.html#index-AC_005fPROG_005fCC_005fC99-885

But you forgot to check result of that macro via ac_cv_prog_cc_c99.

Reply:
Implemented here: 2f8c69a